### DISCUSSION TITLE
BTCPay Server cluster

### DISCUSSION THREAD

**Discussion by rgaitan1982** [btcpayserver/btcpayserver#6492] [Technical Support]
Is it possible to configure BTCPay Server in cluster mode?

I can use haproxy for balancing request to several BTCPay Server instances but, how can I use the same mysql (or postresql) cluster database for all of them instead of local database in each one?

**Comment by Th3-M4jor** [comment]
I asked something similar here https://github.com/btcpayserver/btcpayserver/discussions/6117 as I wanted to scale horizontally for high availability rather than performance reasons and its just not possible under the current architecture of both BTCPayServer and NBXplorer.

**Comment by adriankae** [comment]
tbh this is a problem. 

BTCPay is (supposed to be) critical infrastructure as a payment processor for many merchants. Especially for online businesses that run 24/7.
Server downtime occurs in general (more or less often). I'm just thinking about restarts, some maintenance tasts, etc.

It's actually really hard to understand (and accept) that there isn't a failover solution for BTCPay server since it seems like a mature project (and is working fine as a single instance).

**Reply by Th3-M4jor** [reply]
Yup, NBXPlorer and BTCPayServer might be performant enough that you shouldn't ever have to horizontally scale for performance reasons, but needing / wanting to do so anyway for redundancy (Like if you're running them in k8s) is a fairly common practice.

I had to settle for creating 3 separate instances and having a methodology for taking them out of route as needed for restarts and the like.
